scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py

```diff
@@ -203,7 +207,8 @@ def _handle_table_cell_changed(self, row, col):
                     self.plotted.discard(ws_name)
             elif col == 3:
                 # subtract bg col
-                if is_sub or self.view.get_item_checked(row, 2):  # this ensures the sub ws isn't made on load
+                self.model.update_bgsub_status(ws_name, is_sub)
+                if is_sub or is_plotted:  # this ensures the sub ws isn't made on load
```

Comment:
Can I just check the "or" logic here. The comment says "ensure the sub ws isn't made on load" but this does seem to be what happens. The background subtracted ws is created on load.
If the logic is correct and this is an "or" then perhaps the comment needs updating?

Reply:
Yes I meant to remove the comment - thanks for spotting this!